nixos/modules/system/boot/stage-1.nix

```diff
@@ -510,13 +512,18 @@ in
     };
 
     boot.initrd.compressor = mkOption {
-      internal = true;
-      default = "gzip -9n";
-      type = types.str;
+      default = "gzip";
+      type = types.enum [ "cat" "gzip" "bzip2" "xz" "lz4" "lzop" "zstd" ];
```

Comment:
I'm not all that keen on limiting the options like this, nor on then having to special-case cat elsewhere — e.g. `lzma` is missing here, as well as parallel implementations of some of the algorithms like `pigz` and `pixz`; and the kernel might grow new compression algorithm support. Duplicating that information here isn't ideal.

Reply:
Re: lzma see https://github.com/NixOS/nixpkgs/pull/96005#discussion_r475123132